source/restrict_responses.md

```diff
@@ -1,104 +1,105 @@
-* Restrict a currency input of three digits with a currency sign (either dollar or pound) in front (e.g. **$999** or **£500**): **regex(., '^[\$|\£]\d{3}$')**
-* Restrict an exact input of number of words (e.g. to restrict exactly 3 words "_I love you._"): **regex(., '^\W*(\w+\b\W*){3}$')**
-* Restrict an input of number of words (e.g. to restrict a range of words say 3 to 5): **regex(., '^\W*(\w+\b\W*){3,5}$')**
+* Restrict an input of the beneficiaries name where the initials of the first name and last name are uppercase e.g. _Kobe Bryant_: **regex(., '^[A-Z]{1}[a-z]{1,}[ ]{1}[A-Z]{1}[a-z]{1,}$')**
+* Restrict an input of the beneficiaries name with first name, middle name (if any) and last name e.g. _Kobe Bean Bryant_: **regex(., '^\w{1,}\s(\w{1,})?(\s)?\w{1,}$')**
+* Restrict an input of the beneficiaries' full name where the initials of the names are in uppercase and the name are quite long (often greater than 3 words) e.g. _Samayamantri Venkata Rama Naga Butchi Anjaneya Satya Krishna Vijay_ (this is an example of south Indian names): **regex(., '^([A-Z]{1}[a-z]{1,}\s)([A-Z]{1}[a-z]{1,}\s?)+$')**
+* Restrict an input of the beneficiaries' first name (so that you are able to capture the exact spelling) where the enumerators are forced to enter the beneficiaries first name twice e.g. _Kobe Bryant Kobe_. (This could be helpful when you are trying to document beneficiaries details where a typo error could cost you heavy): **regex(., '^(\D+)\s(\D+)\s?\1$')**
+* Restrict an input of the beneficiaries' last name (so that you are able to capture the exact spelling) where the enumerators are forced to enter the beneficiaries last name twice e.g. _Kobe Bryant Bryant_. _(This could be helpful when you are trying to document beneficiaries details where a typo error could cost you heavy)_: **regex(., '^(\D+)\s(\D+)\s?\2$')**
+* Restrict a character within a word by using the _?_ (quantifier) e.g. allow either _color_ or _colour_ as an input: **regex(., '^colou?r$')**
+* Restrict a character within a word by using the _*_ (quantifier) e.g. allow either _a!_ or _ah!_ or _ahh!_ or _ahhh!_ and so on as an input: __regex(., '^ah*!$')__
+* Restrict a character within a word by using the _+_ (quantifier) e.g. allow either _ah!_ or _ahh!_ or _ahhh!_ and so on as an input: **regex(., '^ah+!$')**
+* Restrict an input to a non-digit character (e.g. _a_ or _c_ or _!_ or _#_ or _%_ etc.): **regex(., '^\D$')**
+* Restrict an input to five non-digit character (e.g. _aZcB!#%_ etc.): **regex(., '^\D{5 }$')**
+
+#### Examples related to use of a combination of alphabets letters and numbers
+
+* Restrict one character which matches between _a_ to _z_ or _A_ to _Z_ or _0_ to _9_ or _ (i.e. match one character from _[a-zA-Z0-9_]_): **regex(., '^\w$')**
```

Comment:
Be careful of the spaces when doing italics, they break the formatting: `_ (i.e. match one character from _[a-zA-Z0-9_]_)` should be `_(i.e. match one character from _[a-zA-Z0-9_])_` (the back ticks are just so that GitHub doesn't format it.

Reply:
To be consistent with what I said above, I would rather you do something like: (_i.e. match one character from_ `[a-zA-Z0-9_]`) — note that this is especially important since the underscore is a valid regex character to include and could confuse the formatting